Draw commodities table using grid or Canvas (to make it scrollable)

mkizub opened the pull request:
Draw table using abstract class that has 2 implementations - old one, using tk.Label and grid layout manager.
And another one that uses Canvas.
The Canvas table is scrollable, and can be resized by mouse drag.
Also fixed some localization.
ViewMode and SortingMode changed to use drop-down menu.

It's not fully tested yet - I did quick tests, but found a couple of minor ussues while playing during a few hours.
I will continue to test it now.

mkizub committed:
Draw commodities table using grid or using Canvas (to make it scrollable)

mkizub commented:
![image](https://github.com/user-attachments/assets/f61d47d9-3ab6-4511-b1b1-f4c1da82302f)

mkizub committed:
Added filter to show only required commodities and added load market data on landing for available commodities; minor style fixes

karolnowacki commented:
cool cool cool, just if you could merger develop and resolve conflicts. I'm preparing to start versioning and will keep main on last release. Every change will be merged to develop and there tested. When we will happy with it we will release to main.
Probably implement auto-updater for it.

mkizub committed:
Merge remote-tracking branch 'karolnowacki/develop' into scrollable-table

# Conflicts:
#	colonization/colonization.py
#	colonization/preferencesui.py
#	colonization/ui.py

mkizub committed:
Add abstract method annotations

mkizub committed:
Fix pylint warnings

mkizub committed:
Fix pylint warnings

mkizub commented:
I've merged and made quick tests. Fixed some pylint warnings in the code. Please, review.
I'll continue testing in real world to make sure I've renamed all those variables correctly.
